### PR TITLE
[perf] Vectorize text-only extend mrope position computation

### PR DESCRIPTION
**Pull request by lcz1998**
Superseded by #30404.

### CODE REVIEW

**Comment by lcz1998** [comment]
Differential-test reproduction script for the accuracy claims in the description (extracts the original `_compute_mrope_positions` verbatim from `origin/main` via AST and compares both implementations end-to-end):

<details>
<summary>difftest_mrope.py (run from repo root: <code>python difftest_mrope.py</code>)</summary>

```python
#!/usr/bin/env python3
"""Differential test: original main-branch `_compute_mrope_positions` vs the
vectorized one, executed end-to-end (including the downstream cat + .to).

The "old" implementation is extracted verbatim from `git show origin/main:...`
via AST, so we compare against the real original code, not a re-typed copy.
"""
import ast
import random
import subprocess
import textwrap
from types import SimpleNamespace

import torch

from sglang.srt.model_executor.forward_batch_info import ForwardBatch, ForwardMode
from sglang.srt.model_executor import forward_batch_info as fbi_module
from sglang.srt.server_args import ServerArgs, set_global_server_args_for_scheduler

set_global_server_args_for_scheduler(ServerArgs(model_path="dummy"))
MODEL_RUNNER = SimpleNamespace(device="cpu")

# -- extract the original method source from git --
blob = subprocess.check_output(
    ["git", "show", "origin/main:python/sglang/srt/model_executor/forward_batch_info.py"],
    text=True,
)
tree = ast.parse(blob)
old_fn_src = None
for node in ast.walk(tree):
    if isinstance(node, ast.ClassDef) and node.name == "ForwardBatch":
        for item in node.body:
            if isinstance(item, ast.FunctionDef) and item.name == "_compute_mrope_positions":
                old_fn_src = ast.get_source_segment(blob, item)
assert old_fn_src is not None, "original method not found"

ns: dict = {
    "torch": torch,
    "get_global_server_args": fbi_module.get_global_server_args,
    "ModelRunner": object,
    "ScheduleBatch": object,
}
exec(textwrap.dedent(old_fn_src), ns)
old_impl = ns["_compute_mrope_positions"]
new_impl = ForwardBatch._compute_mrope_positions
print(f"old impl extracted from origin/main: {len(old_fn_src)} chars")


def make_mm_input(prompt_len: int, delta: int) -> SimpleNamespace:
    return SimpleNamespace(
        mrope_positions=torch.arange(prompt_len, dtype=torch.int64).unsqueeze(0).repeat(3, 1)
        + delta,
        mrope_position_delta=torch.tensor([[delta]], dtype=torch.int64),
        mrope_position_delta_repeated_cache=None,
    )


def run(impl, mode: ForwardMode, reqs: list) -> torch.Tensor:
    """reqs: list of dict(prefix, extend, mm: None | (prompt_len, delta))"""
    fb = ForwardBatch.__new__(ForwardBatch)
    fb.forward_mode = mode
    if mode == ForwardMode.DECODE:
        fb.seq_lens_cpu = torch.tensor([r["seqlen"] for r in reqs])
        batch = SimpleNamespace(
            multimodal_inputs=[
                make_mm_input(*r["mm"]) if r["mm"] else None for r in reqs
            ],
            extend_lens=None,
            prefix_lens=None,
        )
    else:
        fb.seq_lens_cpu = torch.tensor([r["prefix"] + r["extend"] for r in reqs])
        batch = SimpleNamespace(
            multimodal_inputs=[
                make_mm_input(*r["mm"]) if r["mm"] else None for r in reqs
            ],
            extend_lens=[r["extend"] for r in reqs],
            prefix_lens=[r["prefix"] for r in reqs],
        )
    impl(fb, MODEL_RUNNER, batch)
    return fb.mrope_positions


def compare(tag: str, mode: ForwardMode, reqs: list) -> None:
    a = run(old_impl, mode, reqs)
    b = run(new_impl, mode, reqs)
    assert a.shape == b.shape, f"{tag}: shape {a.shape} vs {b.shape}"
    assert a.dtype == b.dtype == torch.int64, f"{tag}: dtype {a.dtype} vs {b.dtype}"
    assert torch.equal(a, b), f"{tag}: values differ"


n = 0
# 1) exhaustive small grid: text-only extend, bs=1
for prefix in range(0, 9):
    for length in range(1, 9):
        compare(f"grid p={prefix} l={length}", ForwardMode.EXTEND,
                [{"prefix": prefix, "extend": length, "mm": None}])
        n += 1

# 2) random large text-only extend, bs 1..5
rng = random.Random(0)
for i in range(200):
    reqs = [
        {"prefix": rng.randint(0, 200_000), "extend": rng.randint(1, 8192), "mm": None}
        for _ in range(rng.randint(1, 5))
    ]
    compare(f"rand-text {i}", ForwardMode.EXTEND, reqs)
    n += 1

# 3) mixed text-only + multimodal requests in one extend batch
for i in range(100):
    reqs = []
    for _ in range(rng.randint(2, 5)):
        if rng.random() < 0.5:
            reqs.append({"prefix": rng.randint(0, 1000), "extend": rng.randint(1, 512), "mm": None})
        else:
            plen = rng.randint(8, 512)
            # prefill within prompt: prefix+extend <= prompt mrope length
            ext = rng.randint(1, plen)
            reqs.append({"prefix": rng.randint(0, plen - ext), "extend": ext,
                         "mm": (plen, rng.randint(-8, 8))})
    compare(f"rand-mixed {i}", ForwardMode.EXTEND, reqs)
    n += 1

# 4) decode batches (untouched path, sanity)
for i in range(50):
    reqs = [
        {"seqlen": rng.randint(1, 100_000),
         "mm": None if rng.random() < 0.5 else (rng.randint(8, 64), rng.randint(-8, 8))}
        for _ in range(rng.randint(1, 5))
    ]
    compare(f"rand-decode {i}", ForwardMode.DECODE, reqs)
    n += 1

print(f"ALL {n} differential cases bit-exact (values, shape, dtype=int64)")

```

</details>

Output on this branch:

```
old impl extracted from origin/main: 2604 chars
ALL 422 differential cases bit-exact (values, shape, dtype=int64)
```
